tracer/src/Datadog.Trace/ExtensionMethods/DictionaryExtensions.cs

```diff
@@ -128,5 +129,42 @@ public static TV Get<TK, TV>(this Dictionary<TK, TV> map, TK key, Func<TK, TV> c
             map[key] = newVal;
             return newVal;
         }
+
+        /// <summary>
+        /// Checks if two dictionaries contain the same keys and values.
+        /// Note that this method assumes the two dictionaries use the same
+        /// <see cref="IEqualityComparer"/>; using different comparers in each
+        /// dictionary is not supported.
+        /// </summary>
+        public static bool SequenceEqual(
+            this ReadOnlyDictionary<string, string> dict1,
+            ReadOnlyDictionary<string, string> dict2,
```

Comment:
any reason to not doing it in a Generic way?

Reply:
Right now, mostly because it's unnecessary, so extra abstraction doesn't seem worth it 🤷‍♂️